### ISSUE TITLE
run analyzePRF without polynomial regressors

### ISSUE THREAD

**Issue by irisgroen** [kendrickkay/analyzePRF#2]
Hi Kendrick,

I'm using analyzePRF to fit pRFs to ECoG data and would like to fit the model without including any polynomials. But I'm not sure how to set the options.maxpolydeg field to achieve this: 

If I leave it empty, it will automatically create some set of polynomials based on the run length.
If I set it to 0, it will still use a single polynomial of degree zero (i.e. a constant, the mean?) 
If I put 'NaN', which according to the help of fitnonlinearmodel.m is the 'special case' without any polynomials being added to the model, I get the following error:

*** analyzePRF: started at 08-Jul-2019 14:39:35. ***
using the following maximum polynomial degrees: [NaN NaN]
constructing seeds.
generating super-grid time-series...done.Elapsed time is 3.064979 seconds.
Error using olsmatrix (line 47)
Assertion failed.
Error in projectionmatrix (line 33)
f = eye(size(X,1)) - X*olsmatrix(X);
Error in analyzePRFcomputesupergridseeds (line 101)
pmatrix = projectionmatrix(blkdiag(pregressors{:}));
Error in analyzePRF (line 335)
  [supergridseeds,rvalues] = analyzePRFcomputesupergridseeds(res,stimulus,data,modelfun, ...

To me it seems like putting 'NaN' as the opt.maxpolydeg field for fitnonlinearmodel.m results in columns of zeros being added as regressors, but there isn't a similar conversion in analyzePRFcomputesupergridseeds.m, which doesn't accept NaNs. 

Thanks for your help!

**Comment by kendrickkay** [comment]
Hey Iris,

Yes, currently, the minimum number of polynomials is 1 (corresponding to the case of a degree zero polynomial, i.e. a constant regressor).  Note that the constant regressor is not necessarily equivalent to the "mean" of the data; rather the code tries to fit a weight on the constant regressor, that combined with the rest of the model, optimally fits the data.

Yeah, you discovered that fitnonlinearmodel.m is a bit more flexible than analyzePRF.m.   Currently, analyzePRF does not support the NaN case.

Is it possible for you to get away to using a constant regressor?   As long as you have some trials in which no stimulus is present, it seems you do have "baseline periods" and it does seem that you should have enough statistical power to go ahead and estimate the weight on the constant regressor.  If so, maybe that's the way forward.    It could be that maybe you already "baseline corrected" your data, in which case the estimation of a constant regressor is a bit redundant, but I don't think it's going to hurt that much (and sanity check would suggest that the weight on the constant regressor should come out to near zero).

Would be happy to chat if you have some things to discuss along these lines..

Kendrick

**Comment by irisgroen** [comment]
Thanks for the quick reply. OK I understand now that 'no additional regressors' is not an option currently in analyzePRF (i.e. this is not a bug). So I'll close this issue.

For my purposes, I do think a constant regressor should be OK, although I have indeed 'baseline corrected' the data, which gives me near-zero activity estimates for the 'baseline periodes' (blanks) in the PRF run. But in my data the pRF estimates (angle, eccentricity) came out quite differently based whether or not I performed the baseline correction, which made me wonder about how the baseline correction interacted with the inclusion of the constant. But as you say I can probably get some more insight into this by inspecting the weight on the constant. Is that one of the params that is outputted as part of the results struct? Or should I read out this weight from somewhere else?
